dip-0008.md

```diff
@@ -275,6 +277,109 @@ resolve any ambiguities which might occur in the last few blocks.
 If the need arises to include block signatures in initial block download, we
 will update this DIP and implementations accordingly.
 
+## Calculations
+
+We consider the scenario where an attacker has assumed control of a number of masternodes. We will then calculate two probabilities that the attacker will
+be able to disrupt the network. Specifically, we will calculate the probability
+that an attacker will be able to prevent all ChainLocks for a twelve hour period, and
+we will calculate the probability that an attacker can successfully create a malicious
+or erroneous ChainLock.
+
+Quorum selection is designed to provide for an effectively simple random
+choice of masternodes in each quorum. Thus we may assume that each quorum is
+a simple random selection. If there are a total of 'N' masternodes the
+number of possible quorums is the number of ways that 400 objects can be
+chosen out of `N`. In some literature this number is
+written <sub>`N`</sub>`C`<sub>`400`</sub> and is enunciated as
+'N choose 400'. Numbers that arise in this fashion are all
+binomial coefficients and have a delightful relation to Pascal's Triangle.
+
+For any process with an observable outcome the set of all outcomes is
+called a sample space.  If all outcomes are equally likely, as in the
+selection of a quorum, the probability of any event is the number of
+outcomes that satisfy that event over the total number outcomes.
+
+An attacker can temporarily prevent ChainLocks if they control
+160 out of 400 masternodes in the quorum.  If the attacker controls
+`m` out of the `N` total masternodes we can compute the
+number of outcomes that satisfy the event "exactly 160
+nodes of the quorum are controlled by the masternode."
+Specifically there are
+<sub>`160`</sub>`C`<sub>`m`</sub> * <sub>`240`</sub>`C`<sub>N - m</sub>
+outcomes where the attacker controls exactly `160` masternodes.
+Thus, the probability that the attacker controls exactly `160` nodes is
+
+<p align="center"><img src="dip-0008/ThreshCase.gif" alt="Threshold Probability"></p>
+
+The attacker could also be successful if they control <b>more</b> that `160`
+of the nodes in the quorum selection. Thus the exact probability that an
+attacker can successfully prevent ChainLocks for that quorum round is
+computed by counting the number of outcomes where the attacker controls
+160 or more nodes in the quorum divided by the total number of outcomes.
+This is computed by the sum
+
+<p align="center"><img src="dip-0008/fin_sum.gif" alt="Final sum" ></p> 
+  
+The numbers involved in this sum can be quite large. We may
+use a <A HREF="dip-0008/quorum_attack.py">Python Script</A> to effectively
+evaluate this sum. Note
+that we made the necessary modifications to this formula
+to calculate the probability that an attacker controls 60% or more
+of the quorum.
+
+| N | m |Type of Attack|success probability in one trial |
```

Comment:
It would probably be better to "squash" the data and restructure this table in another way:
```suggestion
| N | m | Success probability of withholding a ChainLock | Success probability of creating a malicious ChainLock |
```
i.e. the table itself would look like this:
```
| 5000 | 500 | 2.79e-64 | 7.11e-157 |
| 5000 | 1000 | 5.09e-22 | 2.86e-76 |
| 5000 | 1500 | 5.59e-6 | 1.89e-38 |
| 2000 | 200 | 0 | 0 |
| 2000 | 400 | 6.88e26 | 9.49e-94 |
| 2000 | 600 | 1.12e-6 | 3.93e-45 |
```
which IMO is easier to grasp.

Reply:
I like this and I consider this clearer.

Reply:
I need to check the math.  Now that I'm looking at it the change of withholding a chainlock with 200 out of 2000 nodes is non zero, but less that 10e-25

Reply:
*chance of withholding